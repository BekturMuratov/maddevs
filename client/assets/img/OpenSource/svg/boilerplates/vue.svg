<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>