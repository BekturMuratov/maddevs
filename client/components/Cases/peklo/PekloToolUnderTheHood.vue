<template>
  <section class="container_full background-color-black m-48_bottom p-48_top p-48_bottom media-p-41_top media-p-41_bottom">
    <div
      class="container_regular"
      data-aos="fade-up"
      data-aos-anchor-placement="bottom"
    >
      <h2 class="case_title_h2 case_title m-12_bottom">
        Peklo Tool under the hood
      </h2>
      <TextParagraph class="case_bold m-24_bottom">
        Peklo Tool’s architecture features three main components:
      </TextParagraph>
      <div
        class="case_cards m-48_bottom media-m-38_bottom"
      >
        <div
          v-for="card in pekloToolUnderTheHoodCards"
          :key="card.title"
          class="case_card background-color-gray"
        >
          <div class="case_card-text-wrapper">
            <h3
              class="case_title_h5 case_card-title m-8_bottom"
              v-html="card.title"
            >
              {{ card.title }}
            </h3>
            <TextParagraph
              class="case_card-description"
              color="#cfcfd0"
              v-html="card.description"
            >
              {{ card.description }}
            </TextParagraph>
          </div>
          <div class="case_icon-wrapper">
            <img
              v-lazy-load
              :data-src="$getMediaFromS3(`/images/Cases/peklo/svg/${card.icon}.svg`)"
              :alt="card.title"
              :width="card.iconWidth"
              :height="card.iconHeight"
              :class="`case_card-icon-${card.icon}`"
              class="case_card-icon"
            >
          </div>
        </div>
      </div>
      <TextParagraph
        color="#a0a0a1"
        class="case_architecture-description m-24_bottom m-auto"
      >
        Other components deal with payments, file and data storage, and load. Structurally, Peklo Tool’s architecture looks like this:
      </TextParagraph>
      <img
        v-lazy-load
        :data-src="$getMediaFromS3(`/images/Cases/peklo/svg/peklo-tool-under-the-hood.svg`)"
        alt="Peklo Tool’s architecture"
        width="816"
        height="408"
        class="case_architecture-img"
      >
    </div>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'
import { pekloToolUnderTheHoodCards } from '@/data/casePeklo'

export default {
  name: 'PekloToolUnderTheHood',
  components: {
    TextParagraph,
  },

  data() {
    return {
      pekloToolUnderTheHoodCards,
    }
  },
}
</script>

<style lang="scss" scoped>
.case {
  &_title,
  &_bold,
  &_card-title,
  &_card-description,
  &_architecture-description {
    text-align: center;
  }

  &_title,
  &_bold,
  &_card-title {
    color: $text-color--white;
  }

  &_bold {
    font-weight: 700;
  }

  &_architecture-description {
    max-width: 623px;
  }

  &_architecture-img {
    width: 100%;
    height: 100%;
  }

  &_cards {
    @include grid(repeat(3, 1fr), auto, 24px, 0);
  }

  &_card,
  &_icon-wrapper {
    display: flex;
  }

  &_card {
    flex-direction: column;
    justify-content: space-between;
    position: relative;
    border-radius: 10px;
    overflow: hidden;
  }

  &_card-text-wrapper {
    margin-bottom: 20px;
    padding: 24px 10px 0;
  }

  &_icon-wrapper {
    justify-content: flex-end;
  }

  &_card-icon-vue {
    position: relative;
    right: -5px;
  }

  @media screen and (max-width: 990px) {
    &_cards {
      @include grid(repeat(1, 1fr), auto, 0, 24px);
    }

    &_card-title,
    &_card-description {
      /deep/ br {
        display: none;
      }
    }
  }
}
</style>
