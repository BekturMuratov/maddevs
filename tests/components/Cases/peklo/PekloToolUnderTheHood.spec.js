import PekloToolUnderTheHood from '@/components/Cases/peklo/PekloToolUnderTheHood'
import { render } from '@testing-library/vue'

const directives = {
  'lazy-load': () => {},
}

describe('PekloToolUnderTheHood component', () => {
  it('should render correctly', () => {
    const { container } = render(PekloToolUnderTheHood, {
      mocks: {
        $getMediaFromS3: () => 's3 image url',
      },
      directives,
    })

    expect(container).toMatchSnapshot()
  })
})
