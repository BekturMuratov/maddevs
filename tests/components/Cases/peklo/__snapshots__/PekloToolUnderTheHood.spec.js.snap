// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PekloToolUnderTheHood component should render correctly 1`] = `
<div>
  <section
    class="container_full background-color-black m-48_bottom p-48_top p-48_bottom media-p-41_top media-p-41_bottom"
  >
    <div
      class="container_regular"
    >
      <h2
        class="case_title_h2 case_title m-12_bottom"
      >
        
      Peklo Tool under the hood
    
      </h2>
       
      <p
        class="case_paragraph case_bold m-24_bottom"
      >
        
      Peklo Tool’s architecture features three main components:
    
      </p>
       
      <div
        class="case_cards m-48_bottom media-m-38_bottom"
        data-aos="fade-up"
      >
        <div
          class="case_card background-color-gray"
        >
          <div
            class="case_card-text-wrapper"
          >
            <h5
              class="case_title_h5 case_card-title m-8_bottom"
            >
              Go Service for campaign generation
            </h5>
             
            <p
              class="case_paragraph case_card-description"
              style="color: rgb(207, 207, 208);"
            >
              Go puts dictionaries in RAM, ensuring high speed of keyword processing
            </p>
          </div>
           
          <div
            class="case_icon-wrapper"
          >
            <img
              alt="Go Service for campaign generation"
              class="case_card-icon media_lazy case_card-icon-go"
              data-src="s3 image url"
              height="112"
              width="221"
            />
          </div>
        </div>
        <div
          class="case_card background-color-gray"
        >
          <div
            class="case_card-text-wrapper"
          >
            <h5
              class="case_title_h5 case_card-title m-8_bottom"
            >
              Ruby on Rails for backend
            </h5>
             
            <p
              class="case_paragraph case_card-description"
              style="color: rgb(207, 207, 208);"
            >
              it deals with business logic, data management, and admin functionality
            </p>
          </div>
           
          <div
            class="case_icon-wrapper"
          >
            <img
              alt="Ruby on Rails for backend"
              class="case_card-icon media_lazy case_card-icon-ruby"
              data-src="s3 image url"
              height="133"
              width="133"
            />
          </div>
        </div>
        <div
          class="case_card background-color-gray"
        >
          <div
            class="case_card-text-wrapper"
          >
            <h5
              class="case_title_h5 case_card-title m-8_bottom"
            >
              Vue.js 
              <br />
               for frontend
            </h5>
             
            <p
              class="case_paragraph case_card-description"
              style="color: rgb(207, 207, 208);"
            >
              it provides the user interface for the service’s 
              <br />
               clients
            </p>
          </div>
           
          <div
            class="case_icon-wrapper"
          >
            <img
              alt="Vue.js <br> for frontend"
              class="case_card-icon media_lazy case_card-icon-vue"
              data-src="s3 image url"
              height="110"
              width="211"
            />
          </div>
        </div>
      </div>
       
      <p
        class="case_paragraph case_architecture-description m-24_bottom m-auto"
        style="color: rgb(160, 160, 161);"
      >
        
      Other components deal with payments, file and data storage, and load. Structurally, Peklo Tool’s architecture looks like this:
    
      </p>
       
      <img
        alt="Peklo Tool’s architecture"
        class="case_architecture-img media_lazy"
        data-src="s3 image url"
        height="408"
        width="816"
      />
    </div>
  </section>
</div>
`;
